src/meltano/core/utils/__init__.py

```diff
@@ -478,42 +477,111 @@ def __init__(self, env_var: str):
     re.VERBOSE,
 )
 
+Expandable = TypeVar("Expandable", str, Mapping[str, "Expandable"])
+
 
 def expand_env_vars(
-    raw_value: dict[str, str] | str,
-    env: dict[str, str],
+    raw_value: Expandable,
+    env: Mapping[str, str],
+    *,
     raise_if_missing: bool = False,
-):
-    if isinstance(raw_value, dict):
-        return {
-            key: expand_env_vars(val, env, raise_if_missing)
-            for key, val in raw_value.items()
-        }
-    elif not isinstance(raw_value, str):
+    ignore_if_missing: bool = False,
```

Comment:
wdyt about refactoring these args to use a single flag instead:

1. EmptyStr
2. Raise
3. Ignore

Reply:
I had thought about doing that, but didn't want to break every spot that uses the pattern:

```python
with project_settings_service.feature_flag(
    FeatureFlags.STRICT_ENV_VAR_MODE, raise_error=False
) as strict_env_var_mode:
    expand_env_vars(..., raise_if_missing=strict_env_var_mode)
```

`ProjectSettingsService.feature_flag` returns a bool which is passed in directly. We could map that bool to the appropriate enum value at every call site, but that's a bit messy.

One hacky approach we could take is to make it an int enum, have 0 correspond to the default behaviour, and have 1 correspond to raising if missing. That way a bool passed in would work without any changes. What do you think about that @edgarrmondragon?

Reply:
@WillDaSilva Yeah good point, quick search reveals 16 uses of `raise_if_missing=...` 😬 . I'm ok with the hack, with a note to refactor later perhaps?

Reply:
@edgarrmondragon Implemented in cf2333d95a0c51a17173ca9b9fd880903ffd59fa. I'm not sure if it's an improvement or not. If you have strong feelings one way or the other we can keep or revert the changes.

Reply:
@WillDaSilva I think it's a good compromise to avoid having mutually exclusive params 👍